bio/ensembl/ontology/hive/OLSOntologyLoader.py

```diff
@@ -15,29 +14,39 @@
 # @author Marc Chakiachvili
 import logging
 
-from bio.ensembl.ontology.loader.db import dal
-from .OLSHiveLoader import OLSHiveLoader
+import eHive
+from eHive import JobFailedException
 
-logger = logging.getLogger(__name__)
+from bio.ensembl.ontology.loader.db import dal
+from bio.ensembl.ontology.loader.ols import OlsLoader
+from . import param_defaults, log_levels
 
 
-class OLSOntologyLoader(OLSHiveLoader):
+class OLSOntologyLoader(eHive.BaseRunnable):
     """ Dedicated loader for Ontology main page"""
 
     def run(self):
-        # False => erreur marque le job en failed, i.e pas de retry
+        options = param_defaults()
+        options['wipe'] = self.param('wipe_one')
         self.input_job.transient_error = False
+        options['ols_api_url'] = self.param('ols_api_url')
+        options['page_size'] = self.param('page_size')
+        options['output_dir'] = self.param('output_dir')
+        ols_loader = OlsLoader(self.param_required('db_url'), **options)
+        # TODO update options with loader params
+        logging.basicConfig(level=log_levels.get(self.param('verbosity'), logging.ERROR),
+                            datefmt='%m-%d %H:%M:%S')
+        logger = ols_loader.get_ontology_logger(self.param_required('ontology_name'))
         logger.info('Loading ontology info %s', self.param_required('ontology_name'))
         if self.param_required('wipe_one') == 1:
-            self.ols_loader.wipe_ontology(self.param_required('ontology_name'))
-
-        assert self.param_required('ontology_name').upper() in self.ols_loader.allowed_ontologies
+            logger.info("Wiping existing ontology data %s", self.param_required('ontology_name'))
+            ols_loader.wipe_ontology(self.param_required('ontology_name'))
+        try:
+            assert self.param_required('ontology_name').upper() in ols_loader.allowed_ontologies
+        except AssertionError:
+            raise JobFailedException("Ontology %s not implemented" % self.param_required('ontology_name'))
```

Comment:
Why not simply:
```suggestion
        if not self.param_required('ontology_name').upper() in ols_loader.allowed_ontologies:
            raise JobFailedException("Ontology %s not implemented" % self.param_required('ontology_name'))
```

In general, I would use assertions *only* for checking code invariants (e.g. in a function, asserting an object property before returning it), and raise exceptions for any other case as `AssertionError` is limiting the handling of the exception. See: https://stefan.sofa-rockers.org/2018/04/16/assertions-and-exceptions/

Reply:
Another PR will be submitted for this. I think.